backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/api/impl/ReceiptScheduleBL.java

```diff
@@ -752,4 +755,19 @@ public void setBPartnerAddress(final String address)
 			delegate.setBPartnerAddress_Override(address);
 		}
 	}
+
+	@Override
+	public void updateMovementDateForShipperTransportation(@NonNull final ShipperTransportationId shipperTransportationId, @NonNull final LocalDate movementDate)
+	{
+		final ICompositeQueryUpdater<I_M_ReceiptSchedule> updater = queryBL.createCompositeQueryUpdater(I_M_ReceiptSchedule.class)
+				.addSetColumnValue(I_M_ReceiptSchedule.COLUMNNAME_MovementDate, movementDate);
+
+		final int updateCount = receiptScheduleDAO
+				.createQueryForShipperTransportation(shipperTransportationId)
+				.addEqualsFilter(I_M_ReceiptSchedule.COLUMNNAME_Processed, false)
+				.create()
+				.update(updater);
```

Comment:
what's the reason of using ICompositeQueryUpdater instead of some plain BL method?